### DISCUSSION TITLE
config file on startup

### DISCUSSION THREAD

**Discussion by mahiki** [alexmalins/harlequin-databricks#14] [Ideas]
First, thank you so much for your work on this project! Its looking like it will be my favorite way to connect our databricks instance.

Other connectors like duckdb and trino provide for an init_path configuration, which runs initialization commands at the start of the session. Could we have this for databricks too?

Here is what would be in my initialization script every time I log on:

```sql
-- ~/.harlequin_dbx_init.sql

use catalog my_team_catalog;
set time zone 'America/Los_Angeles';
declare prev_ds date default current_date - interval '1' day;
select * from myproject_schema.audit_error_log order by error_date desc limit 20;
```

**Comment by alexmalins** [comment]
hey @mahiki - this is doable 👍 I'll follow what the duckdb and sqlite adapters do and look for a `.databricksrc` file in the user's home directory and run that as the initialization script if the user doesn't specify a path explicitly via the `--init-path/-i/--init` CLI flag

**Reply by mahiki** [reply]
Awesome! This would add a lot of user ergonomics.

**Reply by alexmalins** [reply]
Hey @mahiki - check out v0.5.1 released today. It adds a new feature to run a config file automatically on startup.

In your case you'd run `harlequin -a databricks --init-path ~/.harlequin_dbx_init.sql ...` to run that start up script.

Or if you put those SQL commands into `~/.databricksrc`, harlequin-databricks will automatically run them on launch without having to provide an `--init-path` CLI arg.
